lib/lightning_web/live/workflow_live/edit.ex

```diff
@@ -1202,7 +1213,14 @@ defmodule LightningWeb.WorkflowLive.Edit do
     case type do
       :jobs ->
         socket
-        |> assign(selected_job: value, selected_trigger: nil, selected_edge: nil)
+        |> assign(
+          has_steps: has_steps?(value.id),
```

Comment:
I think this should be either preloaded, a virtual field or something that would apply to all jobs when they are loaded from the database. This create an n+1 situation, which each job is making a new query to the db everytime _any_ change occurs.

Reply:
Done